MyString.java

```diff
@@ -20,81 +29,127 @@ public static void main(String args[]) {
+        }
+        return result;
     }
-  
     /**
-     * Returns a string of n lowercase letters, selected randomly from 
-     * the English alphabet 'a', 'b', 'c', ..., 'z'. Note that the same
-     * letter can be selected more than once.
-     * 
+     * Returns a string of n lowercase letters, selected randomly from the
+     * English alphabet 'a', 'b', 'c', ..., 'z'. Note that the same letter can
+     * be selected more than once.
+     *
      * Example: randomStringOfLetters(3) can return "zoo"
-     * 
+     *
      * @param n - the number of letter to select
      * @return a randomly generated string, consisting of 'n' lowercase letters
      */
-    public static String randomStringOfLetters(int n) {
-        //// Replace the following statement with your code
-        return null;
-    }
-
+    
+        public static String randomStringOfLetters(int n) {
+            Random random = new Random();
+            String alphabet = "abcdefghijklmnopqrstuvwxyz";
+            String result = "";
+        
+            for (int i = 0; i < n; i++) {
+                int index = random.nextInt(alphabet.length()); // Generate a random index
+                result += alphabet.charAt(index); // Concatenate the character to the result
+            }
+        
+            return result;
+        }
     /**
-     * Returns a string consisting of the string str1, minus all the characters in the
-     * string str2. Assumes (without checking) that str2 is a subset of str1.
-     * Example: remove("meet","committee") returns "comit" 
-     * 
+     * Returns a string consisting of the string str1, minus all the characters
+     * in the string str2. Assumes (without checking) that str2 is a subset of
+     * str1. Example: remove("meet","committee") returns//
+     *
      * @param str1 - a string
      * @param str2 - a string
      * @return a string consisting of str1 minus all the characters of str2
      */
     public static String remove(String str1, String str2) {
-       //// Replace the following statement with your code
-        return null;
-    }
+            String originalStr1 = str1;
+            String charsToRemove = str2;
+            String newString = "";
+        
+            if (charsToRemove == null) {
+                return originalStr1;
+            }
+        
+            for (int i = 0; i < originalStr1.length(); i++) {
+                
+                int countAtToRemove =  countChar(charsToRemove, originalStr1.charAt(i));
+                int countAtOriginal = countChar(originalStr1, originalStr1.charAt(i));
+                int countAtNew = countChar(newString,originalStr1.charAt(i));
+        
+                if (countAtNew <  countAtOriginal - countAtToRemove) {
+                    if (countAtNew < countAtOriginal - countAtToRemove ){
+                        newString +=originalStr1.charAt(i);
+                    }
+                }
+            }
+                return newString;
+            }

```

Comment:
seems like over complication.. 
iterate over original1, search for char current2, remove current char from current2.